packages/firebase_vertexai/firebase_vertexai/example/lib/pages/audio_page.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:firebase_vertexai/firebase_vertexai.dart';
+import '../widgets/message_widget.dart';
+import 'package:record/record.dart';
+import 'package:path_provider/path_provider.dart';
+
+final record = AudioRecorder();
+
+class AudioPage extends StatefulWidget {
+  const AudioPage({super.key, required this.title, required this.model});
+
+  final String title;
+  final GenerativeModel model;
+
+  @override
+  State<AudioPage> createState() => _AudioPageState();
+}
+
+
+
+class _AudioPageState extends State<AudioPage> {
+  ChatSession? chat;
+  final ScrollController _scrollController = ScrollController();
+  final List<MessageData> _messages = <MessageData>[];
+  bool _recording = false;
+
+    @override
+  void initState() {
+    super.initState();
+    chat = widget.model.startChat();
+  }
+
+  Future<void> recordAudio() async {
+    if (await record.hasPermission()) {
+      final dir = Directory(
+          '${(await getApplicationDocumentsDirectory()).path}/libs/recordings');
+
+      // ignore: avoid_slow_async_io
+      if (!await dir.exists()) {
+        await dir.create(recursive: true);
+      }
+
+      String filePath =
+          '${dir.path}/recording_${DateTime.now().millisecondsSinceEpoch}.wav';
+
+      await record.start(
+        const RecordConfig(
+          encoder: AudioEncoder.wav,
+        ),
+        path: filePath,
+      );
+    }
```

Comment:
I think the code would be easier to read if you use early returns, something like:
```
if (!await record.hasPermission()) {
  print("Permission denied")
  return;
}

final dir = Directory(
    '${(await getApplicationDocumentsDirectory()).path}/libs/recordings');
...
```

It removes one level of indentation and makes it easier to debug